lib/openhab/dsl/rules/builder.rb

```diff
@@ -993,12 +995,23 @@ def every(value, at: nil, attach: nil)
         #     run { Security_Lights.on }
         #   end
         #
-        def on_load(attach: nil)
-          # prevent overwriting @on_load_id
-          raise ArgumentError, "on_load can only be used once within a rule" if @on_load_id
+        def on_load(delay: nil, attach: nil)
+          # prevent overwriting @on_load
+          raise ArgumentError, "on_load can only be used once within a rule" if @on_load
 
-          @on_load_id = SecureRandom.uuid
-          attachments[@on_load_id] = attach
+          @on_load = { module: SecureRandom.uuid, delay: delay }
+          attachments[@on_load[:module]] = attach
+        end
+
+        # @!visibility private
+        def process_on_load
```

Comment:
any reason not to make this method truly private?

Reply:
oops, my bad! moved to private now.